File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.c
#include <stdio.h>
#include <malloc.h>

typedef struct node
{
	int data;
	struct node *lptr, *rptr;
}node;

node *newnode(int data)
{
	node *newn;
	newn = (node*)malloc(sizeof(node));
	newn->data = data;
	newn->lptr = NULL;
	newn->rptr = NULL;
	return newn;
}

/*Function for PREORDER Transversal*/
void preorder(node *root)
{
	if(root != NULL)
	{
		printf("%d ", root->data);
		preorder(root->lptr);
		preorder(root->rptr);
	}
}

int main()
{
	node *root;
	root = newnode(1);
	root->lptr = newnode(2);
	root->rptr = newnode(3);
	root->lptr->lptr = newnode(4);
	root->lptr->rptr = newnode(5);

	printf("Pre Order Transversal: ");
	preorder(root);

	return 0;
}

File: Algo_Ds_Notes/Tree_Preorder_Traversal/Tree_Preorder_Traversal.py
class node:
    def __init__(self,key):
        self.left = None
        self.right = None
        self.val = key

def PreOrder(root):
    if root:
        print(root.val, end = " ")
        PreOrder(root.left)
        PreOrder(root.right)

root = node(1)
root.left = node(2)
root.right = node(3)
root.left.left = node(4)
root.left.right = node(5)
root.right.left = node(6)
root.right.right = node(7)

print("Pre Order traversal of tree is", end = " ")
PreOrder(root)

''' Output

Pre Order traversal of tree is 1 2 4 5 3 6 7

'''
